ssa/manager_wait.go

```diff
@@ -122,32 +121,36 @@ func (m *ResourceManager) WaitForSet(set object.ObjMetadataSet, opts WaitOptions
 		return statusCollector.Error
 	}
 
-	if ctx.Err() == context.DeadlineExceeded || (opts.FailFast && failedResources > 0) {
-		msg := "failed early due to stalled resources"
-		if ctx.Err() == context.DeadlineExceeded {
-			msg = "timeout waiting for"
-		}
-
-		var errors = []string{}
-		for id, rs := range statusCollector.ResourceStatuses {
-			if rs == nil {
-				errors = append(errors, fmt.Sprintf("can't determine status for %s", utils.FmtObjMetadata(id)))
-				continue
+	var errs []string
```

Comment:
wdyt think about 

```go
 var errs []error
	for id, rs := range statusCollector.ResourceStatuses {
		switch {
		case rs == nil || lastStatus[id] == nil:
			errs = append(errs, fmt.Errorf("resource %s not found", utils.FmtObjMetadata(id)))
		case lastStatus[id].Status == status.FailedStatus:
			var builder strings.Builder
			builder.WriteString(fmt.Sprintf("%s status: '%s'",
				utils.FmtObjMetadata(rs.Identifier), lastStatus[id].Status))
			if rs.Error != nil {
				builder.WriteString(fmt.Sprintf(": %s", rs.Error))
			}
			errs = append(errs, errors.New(builder.String()))
...
	if len(errs) > 0 {
		msg := "failed early due to stalled resources"
		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
			msg = "timeout waiting for"
		}
		return fmt.Errorf("%s: %w", msg, errors.Join(errs...))
	}
```

Wrapping the `errors` can make it easier to inspect for calling funtions.

Reply:
This is more about humans than code, keep in mind that the resulting error ends up in Slack, PagerDuty, etc. Is error wrapping with 100+ errors easier to read and parse than an array?

Reply:
`e.Error()` will return a string still. The difference is that for `Join`. It says `The error formats as the concatenation of the strings obtained by calling the Error method of each element of errs, with a newline between each string.`

Reply:
Ok but this will change the format and if people rely on the current array, their parser will fail. I would consider change it in Flux minor release instead of a patch like now.